pyvista/core/utilities/reader.py

```diff
@@ -2831,6 +2839,255 @@ class ProStarReader(BaseReader):
+        int
+            Number of point arrays.
+        """
+        return self.reader.GetNumberOfPointResultArrays()
+
+    @property
+    def point_array_names(self):
+        """Return the list of all point array names.
+
+        Returns
+        -------
+        list[str]
+            List of all point array names.
+        """
+        return [self.reader.GetPointResultArrayName(i) for i in range(self.number_point_arrays)]
+
+    def enable_point_array(self, name):
+        """Enable point array with name.
+
+        Parameters
+        ----------
+        name : str
+            Point array name.
+
+        """
+        self.reader.SetPointResultArrayStatus(name, 1)
+
+    def disable_point_array(self, name):
+        """Disable point array with name.
+
+        Parameters
+        ----------
+        name : str
+            Point array name.
+
+        """
+        self.reader.SetPointResultArrayStatus(name, 0)
+
+    def point_array_status(self, name):
+        """Get status of point array with name.
+
+        Parameters
+        ----------
+        name : str
+            Point array name.
+
+        Returns
+        -------
+        bool
+            Whether reading the cell array is enabled.
+
+        """
+        return bool(self.reader.GetPointResultArrayStatus(name))
+
+    @property
+    def number_cell_arrays(self):
+        """Return the number of point arrays.
+
+        Returns
+        -------
+        int
+            Number of point arrays.
+        """
+        return self.reader.GetNumberOfElementResultArrays()
+    
+    @property
+    def cell_array_names(self):
+        """Return the list of all cell array names.
+
+        Returns
+        -------
+        list[str]
+            List of all cell array names.
+        """
+
+        return [self.reader.GetElementResultArrayName(i) for i in range(self.number_cell_arrays)]
+
+
+    def enable_cell_array(self, name):
+        """Enable cell array with name.
+
+        Parameters
+        ----------
+        name : str
+            Cell array name.
+
+        """
+        self.reader.SetElementResultArrayStatus(name, 1)
+
+    def disable_cell_array(self, name):
+        """Disable cell array with name.
+
+        Parameters
+        ----------
+        name : str
+            Cell array name.
+
+        """
+        self.reader.SetElementResultArrayStatus(name, 0)
+
+    def cell_array_status(self, name):
+        """Get status of cell array with name.
+
+        Parameters
+        ----------
+        name : str
+            Cell array name.
+
+        Returns
+        -------
+        bool
+            Whether reading the cell array is enabled.
+
+        """
+        return bool(self.reader.GetElementResultArrayStatus(name))
+
+    @property
+    def time_values(self):
+        """All time or iteration values.
+
+        Returns
+        -------
+        list[float]
+
+        """
+
+        vtkStreaming = _lazy_vtk_instantiation("vtkCommonExecutionModel",
```

Comment:
Instead of this method, can the methods like `GetNumberOfTimeSteps` and `Get[Set]TimeStep` be used from `vtkExodusIIReader`?

Reply:
In this case we are getting the time values, whereas I believe GetTimeStep would give you the integer step

Reply:
`TimeReader` already has this implementation for `time_values`

https://github.com/pyvista/pyvista/blob/fadc7e40d860444d73f3e7d026d77457dabb7660/pyvista/core/utilities/reader.py#L658-L666

If `time_point_value` can leverage `GetTimeStep`, then we can get rid of the pipeline here.

Reply:
I looked into this, `GetTimeStep` gets the active time step for reading, so it is not applicable here.